src/ripple/rpc/handlers/UnlLoad.cpp

```diff
@@ -1,45 +0,0 @@
-//------------------------------------------------------------------------------
-/*
-    This file is part of rippled: https://github.com/ripple/rippled
-    Copyright (c) 2012-2014 Ripple Labs Inc.
-
-    Permission to use, copy, modify, and/or distribute this software for any
-    purpose  with  or without fee is hereby granted, provided that the above
-    copyright notice and this permission notice appear in all copies.
-
-    THE  SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
-    WITH  REGARD  TO  THIS  SOFTWARE  INCLUDING  ALL  IMPLIED  WARRANTIES  OF
-    MERCHANTABILITY  AND  FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
-    ANY  SPECIAL ,  DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
-    WHATSOEVER  RESULTING  FROM  LOSS  OF USE, DATA OR PROFITS, WHETHER IN AN
-    ACTION  OF  CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
-    OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
-*/
-//==============================================================================
-
-#include <BeastConfig.h>
-#include <ripple/app/main/Application.h>
-#include <ripple/app/misc/UniqueNodeList.h>
-#include <ripple/core/Config.h>
-#include <ripple/protocol/ErrorCodes.h>
-#include <ripple/net/RPCErr.h>
-#include <ripple/rpc/impl/Handler.h>
-#include <beast/utility/make_lock.h>
-
-namespace ripple {
-
-// Populate the UNL from a local validators.txt file.
-Json::Value doUnlLoad (RPC::Context& context)
-{
-    auto lock = beast::make_lock(context.app.getMasterMutex());
-
-    if (context.app.config().VALIDATORS_FILE.empty ()
-        || !context.app.getUNL ().nodeLoad (context.app.config().VALIDATORS_FILE))
-    {
-        return rpcError (rpcLOAD_FAILED);
-    }
```

Comment:
Yes. We can support that model if we want, but it can make things complicated, especially with manifests: We'd need a way to associate a manifest key to a particular validator, so that when the validator is removed, the associated manifest key is also removed from the list of ephemeral keys.

This may be worthwhile: currently the manifests use a separate configuration stanza `[validator_manifests]` that list the long-term keys. Instead, we could continue using `[validators]` (and `[validators_file]`) and listing the long-term keys there, along with regular validator keys. 

Now, when we receive a manifest, we check if the long-term signing key is listed as a permanent validator and if it is, we add the appropriate ephemeral key, and "link" it to the permanent validator key. Removing one would then remove the other.

Comments?